hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -122,81 +125,120 @@ case class HoodieFileIndex(spark: SparkSession,
+      case Success(opt) => opt
+      case Failure(e) =>
+        logError("Failed to lookup candidate files in File Index", e)
+
+        spark.sqlContext.getConf(DataSkippingFailureMode.configName, DataSkippingFailureMode.Fallback.value) match {
+          case DataSkippingFailureMode.Fallback.value => Option.empty
+          case DataSkippingFailureMode.Strict.value => throw new HoodieException(e);
+        }
+    }
 
     logDebug(s"Overlapping candidate files from Column Stats Index: ${candidateFilesNamesOpt.getOrElse(Set.empty)}")
 
-    var totalFileSize = 0
-    var candidateFileSize = 0
-
-    // Prune the partition path by the partition filters
-    // NOTE: Non-partitioned tables are assumed to consist from a single partition
-    //       encompassing the whole table
-    val prunedPartitions = listMatchingPartitionPaths(partitionFilters)
-    val listedPartitions = getInputFileSlices(prunedPartitions: _*).asScala.toSeq.map {
-      case (partition, fileSlices) =>
-        val baseFileStatuses: Seq[FileStatus] = getBaseFileStatus(fileSlices
-          .asScala
-          .map(fs => fs.getBaseFile.orElse(null))
-          .filter(_ != null))
+    var totalFileSliceSize = 0
+    var candidateFileSliceSize = 0
 
+    val listedPartitions = partitionAndFileSlices.map {
+      case (partitionOpt, fileSlices) =>
         // Filter in candidate files based on the col-stats index lookup
-        val candidateFiles = baseFileStatuses.filter(fs =>
-          // NOTE: This predicate is true when {@code Option} is empty
-          candidateFilesNamesOpt.forall(_.contains(fs.getPath.getName)))
+        val candidateFileSlices: Seq[FileSlice] = {
+          fileSlices.filter(fs => {
+            val baseFileStatusOpt = getBaseFileStatus(Option.apply(fs.getBaseFile.orElse(null)))
+            val logFiles = fs.getLogFiles.collect(Collectors.toSet[HoodieLogFile]).asScala.toSet[HoodieLogFile]
+            // NOTE: This predicate is true when {@code Option} is empty
+            if (candidateFilesNamesOpt.forall(files => baseFileStatusOpt.exists(f => files.contains(f.getPath.getName))
+              || files.intersect(logFiles.map(f => f.getPath.getName)).nonEmpty)) {
+              true
+            } else {
+              false
+            }
+          })
+        }
 
-        totalFileSize += baseFileStatuses.size
-        candidateFileSize += candidateFiles.size
-        PartitionDirectory(InternalRow.fromSeq(partition.values), candidateFiles)
+        totalFileSliceSize += fileSlices.size
+        candidateFileSliceSize += candidateFileSlices.size
+        (partitionOpt, candidateFileSlices)
     }
 
     val skippingRatio =
       if (!areAllFileSlicesCached) -1
-      else if (allFiles.nonEmpty && totalFileSize > 0) (totalFileSize - candidateFileSize) / totalFileSize.toDouble
+      else if (allFiles.nonEmpty && totalFileSliceSize > 0) (totalFileSliceSize - candidateFileSliceSize) / totalFileSliceSize.toDouble
```

Comment:
Is this calculation right? Isn't `allFiles` just a sequence of file satuses of base files.

Reply:
Related to that, should we also have something like `allFiles` - seq of files status of all files for all file slices, and `allBaseFiles` - seq of file status of only base files. Or you can keep `allFiles` only but it would mean seq of base files if `includeLogFiles` is false.

Reply:
The calculation is considering file slices size for calculating skippingRatio.
Yes, allFiles needs to be fixed. Fixed it.